### ISSUE TITLE
AppState removeEventListener is not a function

### ISSUE THREAD

**Issue by RalissonMattias** [microsoft/react-native-code-push#2378]
I'm using sentry and crashlytics to monitor my app and I often get errors on the dashboard, which point to code push

### Steps to Reproduce
1. Create RN project
2. Install and configure last sdk version of the code push 

### Expected Behavior
That the app does not show error logs

### Actual Behavior
Archive mentioned `node_modules/react-native-code-push/CodePush.js`
```
TypeError: f.AppState.removeEventListener is not a function. (In 'f.AppState.removeEventListener("change",t)', 'f.AppState.removeEventListener' is undefined)
```

### Environment

* react-native-code-push version: 7.0.5
* react-native version: 0.70.6
* iOS/Android/Windows version: IOS 16.1
* Does this reproduce on a debug build or release build? Release build
* Does this reproduce on a simulator, or only on a physical device? Physical device

**Comment by trcoffman** [comment]
Yeah, removeEventListener was removed in a recent react-native release. You're supposed to call `.remove()` on the subscription returned by addEventListener now.

https://reactnative.dev/docs/appstate#removeeventlistener

**Comment by RalissonMattias** [comment]
Yeh, that's easy to solve, I saved it to get on their update radar

**Comment by trcoffman** [comment]
This is fixed by #2385, hopefully we can get that reviewed and released soon.

**Comment by RalissonMattias** [comment]
@trcoffman many thanks🚀
